arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -25,176 +181,375 @@ class CommonStressorParams:
         if self.cpu_load is not None:
-            result = result + "cpu-load {}\n".format(self.cpu_load)
+            result += f"cpu-load {self.cpu_load}\n"
+        if self.cpu_method is not None:
+            result += f"cpu-method {CpuMethod(self.cpu_method)}\n"
         return result
 
 
 @dataclass
 class VmStressorParams(CommonStressorParams):
-    vm: int = field(
-        metadata={
-            "name": "VM count",
-            "description": (
-                "Number of VM stressors to be "
-                "run (0 means 1 stressor per CPU"
-            ),
-        }
-    )
-    vm_bytes: str = field(
-        metadata={
-            "name": "VM memory",
-            "description": "Amount of memory a single VM stressor will use",
-        }
-    )
-
-    mmap: typing.Optional[str] = field(
-        default=None,
-        metadata={
-            "name": "mmap",
-            "description": "Number of stressors per CPU",
-        },
-    )
-    mmap_bytes: typing.Optional[str] = field(
-        default=None, metadata={"name": "Allocation of memory per stressor"}
-    )
+    vm_bytes: typing.Annotated[
+        typing.Optional[str],
+        schema.id("vm-bytes"),
+        schema.name("VM Memory Bytes"),
+        schema.description("mmap N bytes per vm worker, the default is 256MB"),
+    ] = None
+
+    vm_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("vm-ops"),
+        schema.name("VM Operations"),
+        schema.description("Stop vm workers after N bogo operations"),
+    ] = None
+
+    vm_hang: typing.Annotated[
+        typing.Optional[int],
+        schema.id("vm-hang"),
+        schema.name("VM Hang"),
+        schema.description(
+            "Sleep N seconds before unmapping memory, the default is zero seconds"
+        ),
+    ] = None
+
+    vm_keep: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-keep"),
+        schema.name("VM Keep"),
+        schema.description(
+            "Do not continually unmap and map memory, just keep on re-writing to it"
+        ),
+    ] = None
+
+    vm_locked: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-locked"),
+        schema.name("VM Locked"),
+        schema.description(
+            "Lock the pages of the mapped region into memory using mmap "
+            "MAP_LOCKED (since Linux 2.5.37)"
+        ),
+    ] = None
+
+    vm_method: typing.Annotated[
+        typing.Optional[VmMethod],
+        schema.id("vm-method"),
+        schema.name("VM Method"),
+        schema.description(
+            "Specify a vm stress method; by default, all the stress methods "
+            "are exercised sequentially"
+        ),
+    ] = VmMethod.ALL
+
+    vm_populate: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-populate"),
+        schema.name("VM Populate"),
+        schema.description(
+            "populate (prefault) page tables for the memory mappings; "
```

Comment:
Most of the descriptions are proper sentences starting with a capital. Some, like this, and `mmap_bytes`, are lowercase phrases. I assume this isn't intentional, and the resulting inconsistency is awkward.

Reply:
It's hard to catch all of these with how big the schema is.